src/main/java/com/mongodb/jdbc/MongoDriver.java

```diff
@@ -99,18 +99,51 @@
 
     @Override
     public Connection connect(String url, Properties info) throws SQLException {
+        Pair<MongoConnection, Integer> p = getUnvalidatedConnectionAndTimeout(url, info);
+        Connection conn = p.left();
+        conn.isValid(p.right());
+        return conn;
+    }
+
+    public MongoConnection createUnvalidatedConnection(String url, Properties info)
+        throws SQLException {
+        return getUnvalidatedConnectionAndTimeout(url, info).left();
+    }
+
+    private Pair<MongoConnection, Integer> getUnvalidatedConnectionAndTimeout(String url, Properties info)
+        throws SQLException {
         if (!acceptsURL(url)) {
             return null;
```

Comment:
I think it would be better to throw an exception here, if we return `null` then the calling functions would have to check for `null` or will throw NPE.

Reply:
No

https://docs.oracle.com/javase/7/docs/api/java/sql/Driver.html#connect(java.lang.String,%20java.util.Properties)

```Attempts to make a database connection to the given URL. The driver should return "null" if it realizes it is the wrong kind of driver to connect to the given URL. This will be common, as when the JDBC driver manager is asked to connect to a given URL it passes the URL to each loaded driver in turn.```

Reply:
Oh okay, good catch find that in the docs.  Then we can return the `null` after line 102 in this file if null, so that lines 103 and on don't fail with the NPE.

Reply:
Oh, yes, duh!